### PR TITLE
rgw build: disable and remove RMDA

### PR DESCRIPTION
**Pull request by m-ildefons**
- disable RDMA features during RGW build
- remove RDMA libraries from s3gw container

fixes: #50
Signed-off-by: Moritz Röhrich <moritz.rohrich@suse.com>

### CODE REVIEW

File: build/Dockerfile.build-container [hunk 1]
```diff
@@ -8,8 +8,6 @@ RUN zypper -n install libblkid1 \
   libexpat1 \
   libtcmalloc4 \
   libfmt8 \
-  libibverbs1 \
-  librdmacm1 \
```
Comment (jecluis) [inline build/Dockerfile.build-container:12]: I'd be curious to see if the resulting container really doesn't have those dependencies installed. I suspect the `install-deps.sh` script will end up pulling them in anyway.

Comment (m-ildefons) [reply build/Dockerfile.build-container:12]: This container does not execute `install-deps.sh`. Also the resuting image was ~60MB smaller, so *some*thing isn't installed.

**Review by jecluis — APPROVED** [review]

**Review by votdev — CHANGES_REQUESTED** [review]
I'm not able to run the container using this PR.
```
~/git/s3gw-core/build   testing ✘ ✭  $ ./build-container.sh
Building container image ...
STEP 1/12: FROM opensuse/tumbleweed
STEP 2/12: LABEL Name=s3gw
--> Using cache 8d36635121269bb50d5317cefe4d0f6572d22c0b2a8ca825f486fc5a028d08d2
--> 8d366351212
STEP 3/12: ARG ID=s3gw
--> Using cache d3c878175ae358af143348a69df078c9d8d3ae0f5471a3f4bfd80ce4de7eb3ec
--> d3c878175ae
STEP 4/12: ENV ID=${ID}
--> Using cache 3503389d686a836f7058a77c3bba1989360d3edc3706df53dff9f2ca209be5c6
--> 3503389d686
STEP 5/12: RUN zypper -n install libblkid1   libexpat1   libtcmalloc4   libfmt8   liboath0   libicu71
Retrieving repository 'openSUSE-Tumbleweed-Non-Oss' metadata [..done]
Building repository 'openSUSE-Tumbleweed-Non-Oss' cache [....done]
Retrieving repository 'openSUSE-Tumbleweed-Oss' metadata [......done]
Building repository 'openSUSE-Tumbleweed-Oss' cache [....done]
Retrieving repository 'openSUSE-Tumbleweed-Update' metadata [..done]
Building repository 'openSUSE-Tumbleweed-Update' cache [....done]
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following 10 NEW packages are going to be installed:
  libblkid1 libexpat1 libfmt8 libicu71 libicu71-ledata liboath0 libtcmalloc4 libunwind oath-toolkit-xml timezone

10 new packages to install.
Overall download size: 11.1 MiB. Already cached: 0 B. After the operation, additional 36.9 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): y
Retrieving package libblkid1-2.37.4-1.4.x86_64 (1/10), 113.7 KiB (215.1 KiB unpacked)
Retrieving: libblkid1-2.37.4-1.4.x86_64.rpm [.done (1.6 MiB/s)]
Retrieving package libexpat1-2.4.8-1.4.x86_64 (2/10),  90.0 KiB (186.1 KiB unpacked)
Retrieving: libexpat1-2.4.8-1.4.x86_64.rpm [done]
Retrieving package libfmt8-8.1.1-2.2.x86_64 (3/10),  84.5 KiB (123.5 KiB unpacked)
Retrieving: libfmt8-8.1.1-2.2.x86_64.rpm [done]
Retrieving package libicu71-ledata-71.1-2.2.noarch (4/10),   8.2 MiB ( 29.0 MiB unpacked)
Retrieving: libicu71-ledata-71.1-2.2.noarch.rpm [..done (4.8 MiB/s)]
Retrieving package libunwind-1.6.2-2.1.x86_64 (5/10),  62.0 KiB (176.6 KiB unpacked)
Retrieving: libunwind-1.6.2-2.1.x86_64.rpm [.done]
Retrieving package oath-toolkit-xml-2.6.7-2.2.noarch (6/10),  16.0 KiB ( 29.0 KiB unpacked)
Retrieving: oath-toolkit-xml-2.6.7-2.2.noarch.rpm [done]
Retrieving package timezone-2022a-2.4.x86_64 (7/10), 445.1 KiB (  1.3 MiB unpacked)
Retrieving: timezone-2022a-2.4.x86_64.rpm [.done (1.2 KiB/s)]
Retrieving package libtcmalloc4-2.9.1-3.1.x86_64 (8/10), 124.9 KiB (299.1 KiB unpacked)
Retrieving: libtcmalloc4-2.9.1-3.1.x86_64.rpm [.done (2.7 KiB/s)]
Retrieving package liboath0-2.6.7-2.2.x86_64 (9/10),  31.7 KiB ( 56.1 KiB unpacked)
Retrieving: liboath0-2.6.7-2.2.x86_64.rpm [done]
Retrieving package libicu71-71.1-2.2.x86_64 (10/10),   1.9 MiB (  5.5 MiB unpacked)
Retrieving: libicu71-71.1-2.2.x86_64.rpm [.done (1.2 KiB/s)]

Checking for file conflicts: [.......done]
( 1/10) Installing: libblkid1-2.37.4-1.4.x86_64 [.........done]
( 2/10) Installing: libexpat1-2.4.8-1.4.x86_64 [.........done]
( 3/10) Installing: libfmt8-8.1.1-2.2.x86_64 [.......done]
( 4/10) Installing: libicu71-ledata-71.1-2.2.noarch [............done]
( 5/10) Installing: libunwind-1.6.2-2.1.x86_64 [.........done]
( 6/10) Installing: oath-toolkit-xml-2.6.7-2.2.noarch [.......done]
( 7/10) Installing: timezone-2022a-2.4.x86_64 [............done]
( 8/10) Installing: libtcmalloc4-2.9.1-3.1.x86_64 [..........done]
( 9/10) Installing: liboath0-2.6.7-2.2.x86_64 [.....done]
(10/10) Installing: libicu71-71.1-2.2.x86_64 [............done]
--> 27e9bcd0271
STEP 6/12: RUN mkdir -p /data
--> 5513e1765ce
STEP 7/12: COPY ./bin/radosgw /usr/bin/radosgw
--> a1544fe8918
STEP 8/12: COPY ./lib/*.so* /usr/lib64/
--> 52c8195eb0d
STEP 9/12: EXPOSE 7480
--> 980395a7e4b
STEP 10/12: VOLUME ["/data"]
--> 65a7e3367e1
STEP 11/12: ENTRYPOINT ["/usr/bin/radosgw", "-d",   "--no-mon-config",   "--id", "${ID}",   "--rgw-data", "/data/",   "--run-dir", "/run/"]
--> f0fffef4c9c
STEP 12/12: CMD ["--rgw-backend-store", "dbstore",   "--debug-rgw", "1"]
COMMIT s3gw
--> af4bb2f1be7
Successfully tagged localhost/s3gw:latest
af4bb2f1be7aa84e81c414933f34e10057e340a3f0078714fce1bd3a06d7770c
```
```
 ~/git/s3gw-core/build   testing ✘ ✭  $ podman images
REPOSITORY                                 TAG         IMAGE ID      CREATED        SIZE
localhost/s3gw                             latest      af4bb2f1be7a  2 minutes ago  323 MB
```
```
 ~/git/s3gw-core/build   testing ✘ ✭  $ podman run --replace --name=s3gw -it -p 7480:7480 localhost/s3gw
cc954ce1471e8d314bec16e89c3e71376d48f530936b05a96f9d995a8f52cd91
/usr/bin/radosgw: error while loading shared libraries: libibverbs.so.1: cannot open shared object file: No such file or directory

```

**Comment by jecluis** [comment]
@votdev did you recompile the code without RDMA?

**Comment by votdev** [comment]
> @votdev did you recompile the code without RDMA?

Arghhh, no. I'm sorry.

**Review by votdev — APPROVED** [review]

**Pull request merged**